tianshou/data/buffer.py

```diff
@@ -181,20 +79,95 @@ def __setstate__(self, state: Dict[str, Any]) -> None:
         We need it because pickling buffer does not work out-of-the-box
         ("buffer.__getattr__" is customized).
         """
-        self._indices = np.arange(state["_maxsize"])
         self.__dict__.update(state)
+        # compatible with previous version HDF5
```

Comment:
what does 'previous version HDF5' means ? Not informative enough to be useful except for the original writer.

Reply:
My bad. 
```
# compatible with version == 0.3.1's HDF5 data format
```